tests/auto/api.json

```diff
@@ -1932,6 +1951,66 @@
+                "baudRate": "Uint",
+                "o:channelMask": "Uint",
+                "serialPort": "String"
+            },
+            "returns": {
+                "o:networkUuid": "Uuid",
+                "zigbeeError": "$ref:ZigbeeError"
+            }
+        },
+        "Zigbee.FactoryResetNetwork": {
+            "description": "Factory reset the network with the given networkUuid. The network does not have to be Online for this procedure, and all associated nodes and things will be removed permanently. Make sure the user realy wants to do this because this can not be undone.",
+            "params": {
+                "networkUuid": "Uuid"
+            },
+            "returns": {
+                "zigbeeError": "$ref:ZigbeeError"
+            }
+        },
+        "Zigbee.GetAdapters": {
+            "description": "Get the list of available Zigbee adapter candidates in order to set up the zigbee network on the desired serial interface. The serialPort property can be used as unique identifier. If an adapter hardware has been recognized as a supported hardware, the 'hardwareRecognized' property will be true and the baud rate and backend configurations can be used as they where given, otherwise the user might set the backend type and baud rate manually.",
+            "params": {
+            },
+            "returns": {
+                "adapters": "$ref:ZigbeeAdapters"
+            }
+        },
+        "Zigbee.GetNetworks": {
+            "description": "Returns the list of zigbee networks configured in the system.",
+            "params": {
+            },
+            "returns": {
+                "zigbeeNetworks": [
+                    "$ref:ZigbeeNetwork"
+                ]
+            }
+        },
+        "Zigbee.RemoveNetwork": {
+            "description": "Remove the zigbee network with the given network uuid.",
+            "params": {
+                "networkUuid": "Uuid"
+            },
+            "returns": {
+                "zigbeeError": "$ref:ZigbeeError"
+            }
+        },
+        "Zigbee.SetPermitJoin": {
+            "description": "Allow or deny nodes to join the network with the given networkUuid for a specific duration in seconds. The duration values has to be between 0 and 255 seconds. The permitJoinDuration indicates how long permit has been enabled and the permitJoinDuration indicates the rest of the time. Those values can be used to show a countdown or progressbar. This method can be recalled for resetting the timeout. If the duration is set to 0 seconds, joining will be disabled immediatly for the entire network. The shortAddress is optional and defaults to the broadcast address (0xfffc) for all routers in the network. If the short address matches the address of a router node in the network, only that router will be able to allow new nodes to join the network. A new node will join to the router with the best link quality index (LQI).",
```

Comment:
"The duration values has", either singular or plural, but not mixed. I guess singular in this case.

The permitJoinDuration indicates how long permit has been enabled and the permitJoinDuration indicates the rest of the time. 

a) there is no permitJoinDuration parameter
b) ... indicates for how long joining may be permitted...
c) second one shouldn't be permitJoinDuration, also it's not a parameter for this method, so should be removed here...

Reply:
How does the CLI close the network after the first join? Does it do that on its own (I.e. watch for nodes and call this with duration 0 as soon as a node appears)? I though that's a feature in the core but can't see how to enable that from this API.